File: AddNewGames.py
import csv
import json

#method will update the csv file
def updateCSV(gameName, gamePath):
    with open(r'C:\Users\Daniel\OneDrive\Desktop\Test.csv', 'a') as file:
        test = [[gameName], [gamePath]]
        ex = csv.writer(file)
        #testing = [(gameName),(gamePath)]
        ex.writerows(test)
        

#method will update json file
def updateJSON(gameName, gamePath):

    with open(r"C:\Users\Daniel\OneDrive\Desktop\Games\GamePathJSON.json") as loadGames:
        newGames = json.load(loadGames)

    newGames[gameName] = gamePath

    with open(r"C:\Users\Daniel\OneDrive\Desktop\Games\GamePathJSON.json", "w") as writeGames:
        json.dump(newGames, writeGames)

#method will update two txt files
def updateTXT(gameName, gamePath):
    pass

#main method that executes the program
def main():
    print("This python program adds a new game and an associated file path to file associated with the PMFG programs. \n")
    gameName = str(input("Please enter the game name: \n"))
    gamePath = str(input("Please enter the game file path: \n"))

    updateCSV(gameName, gamePath)
    #updateJSON(gameName, gamePath)
    #updateTXT(gameName, gamePath)

    print("All files have been updated")    


main()

File: AddNewGames.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class AddNewGames {

    //Input validation method that checks to see if the user inputs an actual file path
    private static String checkPath(String gamePath){
        Scanner scanner = new Scanner(System.in);
        String correctPath = "";
        //Booleans that find the \\ characters in the file path and check for an exe file
        Boolean slash = gamePath.contains("\\");
        Boolean exe = gamePath.contains(".exe");
        //If statement that validates the input, returns correctPath if they input something, other than a filepath.
        if (exe == true && slash == true){
            return gamePath;
        } else{
            while (exe == false || slash == false){
                System.out.println("Please enter a file path that leads to an exe file and has slashes:");
                correctPath = scanner.nextLine();
                Boolean slash2 = correctPath.contains("\\");
                Boolean exe2 = correctPath.contains(".exe");
                if (exe2 == true && slash2 == true) {
                    break;
                }
            }
            return correctPath;
        }

    }

    //method that counts lines in the text file. When this program opens, this method counts the amount of games already in there and adds a corresponding number.
    private static int countLines(String filePath){
        int gameCount = 0;
        try {
            BufferedReader br = new BufferedReader(new FileReader(filePath));
            while (br.readLine() != null) gameCount++;
        }catch (IOException e){
            System.out.println("Can not count lines in text file.");
        } finally{

        }
        return gameCount;

    }

    public static void updateJSON(String gameName, String gamePath){
        ;
        //I must import the appropriate class before i can work on this method, it will update the JSON file on my computer.
    }

    //this method updates the text files on the computer
    public static void updateTXT(String gameName, String gamePath){
        String filePath = "C:\\Users\\Daniel\\OneDrive\\Desktop\\Games\\GameFilePath.txt";
        String fileName = "C:\\Users\\Daniel\\OneDrive\\Desktop\\Games\\GameListTXT.txt";
        try {
            FileWriter gP = new FileWriter(filePath, true);
            FileWriter gN = new FileWriter(fileName, true);
            gN.write("\n" + countLines(fileName) + " - " + gameName);
            gP.write("\n" + gamePath);
            gN.close();
            gP.close();
        } catch (IOException e) {
            System.out.println("This is an invalid filepath, please check your locations and try again");
        }
    }

    public static void updateCSV(String gameName, String gamePath){
        ;
        //I must import the appropriate class before i can work on this method, it will update the CSV file on my computer.
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        System.out.println("This java program adds a new game and an associated file path to files associated with the PMFG\\Game Runner programs.");
        System.out.println("Please enter the game name:");
        String gameName = scanner.nextLine();
        System.out.println("Please enter the game file path:");
        String gamePath = scanner.nextLine();
        String checked = checkPath(gamePath);

        updateTXT(gameName, checked);
        System.out.println("Files updated successfully");
        
    }
}
